source/pigeonhole-man/pigeonhole.7.rst.in

```diff
@@ -0,0 +1,106 @@
+.. _man-7_pigeonhole:
+
+==========
+pigeonhole
+==========
+
+NAME
+====
+
+pigeonhole - Overview of the Pigeonhole project's :ref:`Sieve <sieve>` support for the
+Dovecot secure :ref:`IMAP <imap>` and :ref:`POP3 <pop3>` server.
+
+DESCRIPTION
+===========
+
+Pigeonhole Project (:man:`pigeonhole(7)`) adds support for the
+Sieve language (`RFC 5228 <http://tools.ietf.org/html/rfc5228/>`_) and
+the ManageSieve protocol (`RFC 5804 <http://tools.ietf.org/html/rfc5804/>`_) to
+Dovecot secure :ref:`IMAP <imap>` and :ref:`POP3 <pop3>` server (:man:`dovecot(1)`).
+In the literal sense, a pigeonhole is a a hole or recess inside a dovecot for
+pigeons to nest in. It is, however, also the name for one of a series of
+small, open compartments in a cabinet used for filing or sorting mail.
+As a verb, it describes the act of putting an item into one of those
+pigeonholes. The name "Pigeonhole" therefore well describes an important
+part of the functionality that this project adds to Dovecot: sorting and
+filing e-mail messages.
+
+The Sieve language is used to specify how e-mail needs to be processed.
+By writing Sieve scripts, users can customize how messages are
+delivered, e.g. whether they are forwarded or stored in special folders.
+Unwanted messages can be discarded or rejected, and, when the user is
+not available, the Sieve interpreter can send an automated reply. Above
+all, the Sieve language is meant to be simple, extensible and system
+independent. And, unlike most other mail filtering script languages, it
+does not allow users to execute arbitrary programs. This is particularly
+useful to prevent virtual users from having full access to the mail
+store. The intention of the language is to make it impossible for users
+to do anything more complex (and dangerous) than write simple mail
+filters.
+
+Using the ManageSieve protocol, users can upload their Sieve scripts
+remotely, without needing direct filesystem access through FTP or SCP.
+Additionally, a ManageSieve server always makes sure that uploaded
+scripts are valid, preventing compile failures at mail delivery.
+
+The Pigeonhole project provides the following items:
+
+-  The LDA Sieve plugin for Dovecot's Local Delivery Agent (LDA)
+   (:man:`dovecot-lda(1)`) that facilitates the actual Sieve filtering
+   upon delivery.
+
+-  The ManageSieve service that implements the ManageSieve protocol
+   through which users can remotely manage Sieve scripts on the server.
+
+-  A plugin for Dovecot's :man:`doveadm(1)` command line tool that adds
+   new :man:`doveadm-sieve(1)` commands for management of Sieve
+   filtering.
```

Comment:
The IMAPSieve plugin and FILTER=SIEVE plugin should probably also be mentioned here now.